lib/dot-compat.js

```diff
@@ -0,0 +1,298 @@
+
+    // check for special settings for eslint:all and eslint:recommended:
+    if (eslintrcConfig.settings) {
+        if (eslintrcConfig.settings["eslint:all"] === true) {
+            return ["eslint:all"];
+        }
+
+        if (eslintrcConfig.settings["eslint:recommended"] === true) {
+            return ["eslint:recommended"];
+        }
+    }
+
+    // copy over simple translations
+    for (const key of keysToCopy) {
+        if (key in eslintrcConfig && typeof eslintrcConfig[key] !== "undefined") {
+            dotConfig[key] = eslintrcConfig[key];
+        }
+    }
+
+    // copy over languageOptions
+    for (const key of languageOptionsKeysToCopy) {
+        if (key in eslintrcConfig && typeof eslintrcConfig[key] !== "undefined") {
+            dotConfig.languageOptions = languageOptions;
+
+            if (languageOptions[key] && typeof languageOptions[key] === "object") {
+                languageOptions[key] = {
+                    ...eslintrcConfig[key]
+                };
+            } else {
+                languageOptions[key] = eslintrcConfig[key];
+
+                if (key === "parser") {
+                    debug(`Resolving parser '${languageOptions[key]}' relative to ${resolveConfigRelativeTo}`);
+                    languageOptions[key] = eslintrcConfig[key].definition;
+                }
+            }
+        }
+    }
+
+    // copy over linterOptions
+    for (const key of linterOptionsKeysToCopy) {
+        if (key in eslintrcConfig && typeof eslintrcConfig[key] !== "undefined") {
+            dotConfig.linterOptions = linterOptions;
+            linterOptions[key] = eslintrcConfig[key];
+        }
+    }
+
+    // move ecmaVersion a level up
+    if (languageOptions.parserOptions) {
+
+        if ("ecmaVersion" in languageOptions.parserOptions) {
+            languageOptions.ecmaVersion = languageOptions.parserOptions.ecmaVersion;
+            delete languageOptions.parserOptions.ecmaVersion;
+        }
+
+        if ("sourceType" in languageOptions.parserOptions) {
+            languageOptions.sourceType = languageOptions.parserOptions.sourceType;
+            delete languageOptions.parserOptions.sourceType;
+        }
+
+        // check to see if we even need parserOptions anymore and remove it if not
+        if (Object.keys(languageOptions.parserOptions).length === 0) {
+            delete languageOptions.parserOptions;
+        }
+    }
+
+    if (eslintrcConfig.ignorePattern) {
+        configs.unshift({
+            ignores: eslintrcConfig.ignorePattern.patterns
+        });
+    }
+
+    // overrides
+    if (eslintrcConfig.criteria) {
+
+        for (const { includes, excludes } of eslintrcConfig.criteria.patterns) {
+            if (includes) {
+                dotConfig.files = includes.map(include => include.pattern);
+            }
+
+            if (excludes) {
+                if (!includes) {
+                    dotConfig.files = [excludes.map(exclude => `!${exclude.pattern}`)];
```

Comment:
As it turns out, we actually don't allow `excludedFiles` without `files` (I thought we did, but when I tried to test that, `ConfigArrayFactory` threw an error. I just left this here in case I missed an edge case.